frontend/src/stateless/index.ts

```diff
@@ -0,0 +1,372 @@
+        }
+      };
+
+      request.onsuccess = function (event: any) {
+        const db = event.target.result;
+        const transaction = db.transaction(['kubeconfigStore'], 'readonly');
+        const store = transaction.objectStore('kubeconfigStore');
+
+        store.openCursor().onsuccess = function (event: any) {
+          const cursor = event.target.result;
+          if (cursor) {
+            const kubeconfigObject = cursor.value;
+            const kubeconfig = kubeconfigObject.kubeconfig;
+
+            const parsedKubeconfig = jsyaml.load(atob(kubeconfig)) as KubeconfigObject;
+
+            const matchingKubeconfig = parsedKubeconfig.clusters.find(
+              cluster => cluster.name === clusterName
+            );
+
+            if (matchingKubeconfig) {
+              resolve(kubeconfig);
+            } else {
+              cursor.continue();
+            }
+          } else {
+            resolve(null); // No matching kubeconfig found
+          }
+        };
+      };
+
+      request.onerror = function (event: any) {
+        console.error('Error opening the database:', event.target.error);
+        reject(event.target.error);
+      };
+    } catch (error) {
+      reject(error);
+    }
+  });
+}
+
+/**
+ * In the backend we use a unique ID to identify a user. If there is no ID in localStorage
+ * we generate a new one and store it in localStorage. We then combine it with the
+ * cluster name and this headlamp-userId to create a unique ID for a cluster. If we don't
+ * do it then if 2 different users have a cluster with the same name, then the
+ * proxy will be overwritten.
+ * @returns headlamp-userId from localStorage
+ */
+export function getUserIdFromLocalStorage(): string {
+  let headlampUserId = localStorage.getItem('headlamp-userId');
+
+  if (!headlampUserId) {
+    headlampUserId = generateSecureToken();
+
+    if (headlampUserId) {
+      localStorage.setItem('headlamp-userId', headlampUserId);
+    }
+  }
+
+  return headlampUserId!;
+}
+
+/**
+ * Generates a cryptographically secure random token using the browser's crypto API.
+ * @param {number} length - The length of the token.
+ * @returns {string} - The generated token.
+ */
+function generateSecureToken(length = 16): string {
+  const buffer = new Uint8Array(length);
+  if (process.env.NODE_ENV === 'test') {
+    // Use Math.random() in the testing environment
+    return Array.from(buffer, () => Math.floor(Math.random() * 16).toString(16)).join('');
+  }
+  window.crypto.getRandomValues(buffer);
+  return Array.from(buffer, byte => byte.toString(16).padStart(2, '0'))
+    .join('')
+    .slice(0, length);
+}
+
+/**
+ * Compares the cluster config from the backend and the redux store
+ * and returns true if the present stored config is different from the fetched one.
+ * @param clusters
+ * @param clustersToConfig
+ * @param isStateless
+ */
+export function processClusterComparison(
```

Comment:
I think the e2e has enough coverage for now. I plan to add more e2e tests later once the PR is merged to test edge cases. I think it would be enough to cover the functionality.